### PR TITLE
feat(riviere-extract-ts): implement draft component extraction

### LINKED ISSUE #42: [M3-D3.3] Implement draft component output format
### Context
**PRD:** `docs/project/PRD/active/PRD-phase-10-typescript-extraction.md` — M3 D3.3

Extracted components are output as "draft" JSON—an intermediate format before connection detection. This format contains component identity and location but no relationships.

### Key Decisions and Principles
- **Components only** — This phase extracts component nodes; connections come in PRD 12
- **Draft JSON output** — Not a valid Riviere graph (no connections); format documented in package
- **Domain from config** — Component domain derived from module path in extraction config

### Delivers
The extractor produces valid JSON output with all required fields (type, name, location, domain) for each detected component.

### Acceptance Criteria
- Given source code with components When extraction runs Then JSON output contains all detected components
- Given a detected component When examining output Then it has type, name, file location, and line number
- Given an extraction config with module "orders" at path "src/orders/**" When a component is found in that path Then its domain is "orders"
- Given the output format When validated Then it matches the documented schema

### Dependencies
Depends on #41

### Related Code
- `packages/riviere-extract-ts/` — Where output is generated
- `packages/riviere-schema/` — Reference for component type definitions

### Verification
```bash
nx test riviere-extract-ts
```
- Unit tests for output structure
- Integration test extracting from sample code
- 100% test coverage

Example output:
```json
{
  "components": [
    {
      "type": "API",
      "name": "createOrder",
      "location": { "file": "src/orders/api/orders.controller.ts", "line": 42 },
      "domain": "orders"
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by NTCoding**
### **User description**
Closes #42

## Summary

- Implements the core extraction engine that processes TypeScript source files
- Matches files to module path patterns using minimatch glob matching
- Applies detection rules (find + where predicates) to extract components
- Supports extracting classes, methods, and functions as draft components
- Enriches each component with type, name, location (file, line), and domain
- Integrates all 9 predicates from previous implementation
- Added minimatch dependency for glob pattern matching
- Achieves 100% branch coverage with comprehensive test cases

Closes #42

🤖 Generated with [Claude Code](https://claude.com/claude-code)

___

### **PR Type**
Enhancement

___

### **Description**
- Implements core extraction engine processing TypeScript source files

- Matches files to module patterns using minimatch glob matching

- Applies detection rules to extract classes, methods, and functions

- Enriches components with type, name, location, and domain metadata

- Adds `name` field to Module type for domain identification

- Expands test coverage with 12+ comprehensive extraction scenarios

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Source Files"] -->|"minimatch pattern"| B["Module Matching"]
  B -->|"matched module"| C["Detection Rules"]
  C -->|"find + where predicates"| D["Extract Classes/Methods/Functions"]
  D -->|"enrich metadata"| E["DraftComponents"]
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td><strong>extractor.ts</strong><dd><code>Implement complete extraction engine with AST traversal</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-842bbd635fc507cfb17671b68a96e55a0b1411f893709ca008878b9ec422cf04">+198/-3</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>types.ts</strong><dd><code>Add name field to Module interface</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-9db4ecf4471f1c383e271bfba5843532b4b3d7d590585345fc7344252a92aa1f">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>validation-fixtures.ts</strong><dd><code>Update fixtures to include module name field</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-94364fe5282571c6c1635fe2e0154c3dcc342ce6771f3d0263f8f68f9bfbe2b2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>index.ts</strong><dd><code>Export validation fixture helper functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-879f8d4b855838313064d4dd3a30bad071b17c82e7299e68e7a20f0e00700fc2">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>extractor.spec.ts</strong><dd><code>Add comprehensive test suite for extraction scenarios</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-08a0999139ac91decbe7c9cafa771572e7695a0f3ce327c85be4fb625c1dda5a">+289/-8</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>default-config.spec.ts</strong><dd><code>Update test expectations for module field count</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-ab5fa2e201327c9f6049cd6975558b021124cb6757e0208248adbeb3bb1ab666">+11/-2</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>extraction-config.schema.json</strong><dd><code>Add name field to schema with validation rules</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-8e04f234fa622edf1c11b02d6799398a4f85e15f9b640cfcc22da2af46ef321e">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>default-extraction.config.json</strong><dd><code>Add name field to default module configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-7c546af2e33f2bbaaebe8a7166717e797303a26ee46e9ef2f843fb3f598a37d4">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>settings.json</strong><dd><code>Allow submit-pr.sh script execution permissions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-f27ac6f39d89fe021c56900069198aa7d9968f2cd6645c00b11ffd1b78fcf546">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>test-fixtures.ts</strong><dd><code>Remove duplicate fixture, use shared config</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-45f8ec5771d03b640352633b17c4540ecf6e9bc4fb7e394a5e64259782981ee8">+0/-17</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>package.json</strong><dd><code>Add minimatch dependency for glob pattern matching</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-e198dc3f4997a1bd3d17caf35c09b4f681791f9d3654bf7aa755027f15c38c46">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>pnpm-lock.yaml</strong><dd><code>Update lock file with minimatch and dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/NTCoding/living-architecture/pull/84/files#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bb">+25/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tbody></table>

</details>

___

## Summary by CodeRabbit

* **New Features**
  * Modules now require a mandatory "name" field.

* **Bug Fixes**
  * Improved component extraction with better handling of missing files, path mismatches, and Windows paths; more accurate location reporting.

* **Tests**
  * Expanded test coverage for edge cases and component-type extraction.

* **Documentation**
  * Added "Parse, Don't Validate" guidance for safer boundary parsing.

* **Chores**
  * Added a dependency to support module path matching.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/riviere-extract-ts/src/extractor.ts [hunk 1]
```diff
@@ -12,9 +24,192 @@ export interface DraftComponent {
...
+      type: componentType,
+      name,
+      location: {
+        file: filePath,
+        line: method.getStartLineNumber(),
+      },
+      domain,
+    },
+  ]
+}
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-extract-ts/src/extractor.ts]: Not fixing: MethodDeclaration.getName() always returns string (not string | undefined) because methods cannot be anonymous in TypeScript. Only ClassDeclaration and FunctionDeclaration can have undefined names. The ts-morph type signature confirms this - no guard needed.

🤖 *This is an automated response from [Claude Code](https://claude.ai/claude-code). The repository owner did not write this message.*

[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-extract-ts/src/extractor.ts]: Not fixing: ts-morph's MethodDeclaration.getName() always returns string (not string | undefined). Unlike ClassDeclaration and FunctionDeclaration which can be anonymous, method declarations in TypeScript always have names - there's no such thing as an anonymous method. The type signature confirms this: MethodDeclaration extends NamedNode which has getName(): string.

🤖 *This is an automated response from [Claude Code](https://claude.ai/claude-code). The repository owner did not write this message.*
